napari/components/overlays/interaction_box.py

```diff
@@ -1,93 +1,43 @@
-import numpy as np
+from typing import Optional, Tuple
 
-from napari.utils.events import EventedModel
-from napari.utils.events.custom_types import Array
-from napari.utils.transforms import Affine
+from napari.components.overlays.base import SceneOverlay
+from napari.layers.utils.interaction_box import (
+    InteractionBoxHandle,
+    calculate_bounds_from_contained_points,
+)
 
 
-class InteractionBox(EventedModel):
-    """Models a box that can be used to transform an object or a set of objects
+class SelectionBoxOverlay(SceneOverlay):
+    """A box that can be used to select and transform objects.
 
-    Parameters
+    Attributes
     ----------
-    points : list
-        Nx2 array of points whose interaction box is to be found
-    show : bool
-        Bool indicating whether the box should be drawn
-    show_handle : bool
-        Bool indicating whether the full box with midpoints and rotation handle should be drawn.
-        If False only the corners are drawn.
-    show_vertices : bool
-        Bool indicating whether vertices that enable interaction with the box should be drawn
-    transform : napari.util.transforms.Affine
-        Holds an affine transform that is modified by interaction with the vertices of the box
-    selection_box_drag : list
-        4x2 list of a box that is being drawn to select items. Gets updated during the mouse-drag
-    selection_box_final : list
-        4x2 list of a box that has been drawn to select item. Gets updated once the mouse drag is finished
-    transform_start : napari.util.transforms.Affine
-        The affine transformation of the box before a new mouse drag was started. Gets updated when a new drag begins.
-    transform_drag : napari.util.transforms.Affine
-        The affine transformation of the box during a mouse drag. Gets updated when mouse moves during drag.
-    transform_final : napari.util.transforms.Affine
-        The affine transformation of the box after a mouse drag. Gets updated when a mouse drag is finished.
-    allow_new_selection: bool
-        Bool indicating whether the interaction box allows creating a new selection by dragging outside an existing interaction_box.
+    visible : bool
+        If the box is visible or not.
```

Comment:
I do not see where `visible` is defined. Also, I do not see `opacity`, `order`, and `position` in the docstring.